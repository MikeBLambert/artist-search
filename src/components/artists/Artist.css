#artistInfo {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  font-family: 'Black Ops One', cursive;
  color: black;

}