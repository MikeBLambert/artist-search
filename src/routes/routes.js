import ArtistList from '../components/artists/ArtistList';
import ArtistDetail from '../components/artistDetail/ArtistDetail';
import Song from '../components/song/Song';

export const ROUTES = {
  ARTISTS: {
    path: '/artists',
    Component: ArtistList,
    linkPath: () => '/artists'
  },
  ARTIST: {
    path: '/artists/:id',
    Component: ArtistDetail,
    linkPath: id => `/artists/${id}`
  },
  SONG: {
    path: '/lyrics/:artist/:song',
    Component: Song,
    linkPath: (artist, song) => `/lyrics/${artist}/${song}`
  }
};
