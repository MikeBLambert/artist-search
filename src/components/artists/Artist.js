import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import styles from './Artist.css';

export default function Artist({ id, name }) {


  return (
    <div id={styles.artistInfo}>
      <h3>{name}</h3>
      <Link to={ROUTES.ARTIST.linkPath(id)}>See all songs</Link>
    </div>
  );
}
