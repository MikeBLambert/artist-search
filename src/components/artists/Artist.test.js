import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';
import { MemoryRouter } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';

describe('Artist component', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Artist id='9876'/>);
  });

  it('renders an episode', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays the artist\'s name', () => {
    const wrapperWithoutRouter = shallow(
      <MemoryRouter>
        <Artist id='666' name='Nirvana' />
      </MemoryRouter>
    );
    expect(wrapperWithoutRouter.html()).toContain('Nirvana');
  });

  it('links to an artist detail', () => {
    expect(wrapper.find('Link').props().to).toEqual(ROUTES.ARTIST.linkPath('9876'));
  });
});
