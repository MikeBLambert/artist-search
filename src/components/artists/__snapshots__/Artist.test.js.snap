// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders an episode 1`] = `
<div
  id="artistInfo"
>
  <h3 />
  <Link
    replace={false}
    to="/artists/9876"
  >
    See all songs
  </Link>
</div>
`;
